### PR TITLE
5075 - Add support for mobile view and stay on provided date with monthview

### LINKED ISSUE #5064: Calendar: Today becomes selected day on calendar in current month instead of provided day
**Describe the bug**
Calendar initialized with day in current month will select today instead of provided day.

**To Reproduce**

**Steps to reproduce the behavior:**
_example-only-calendar.html:_
`        $('.calendar').calendar({
          day: 21,
          month: 3,
          year: 2021, ... });`
(Today being any day in April 2021 besides the 21st)

Create calendar using object containing year, month, day in current month where day isn't today's date.

This will be shown on April 9, 2021:
image

**Expected behavior**
The day provided to calendar should be highlighted/selected.
image

**Version**
Tested in ids-enterprise v4.38.x and v4.50.x.

**Platform**
 - Infor Application/Team Name: Infor Pathway
 - Browser Name: chrome
 - Browser Version: Version 89.0.4389.114 (Official Build) (64-bit)

### LINKED ISSUE #5075: DatePicker: Add Mobility enhancements
**Is your feature request related to a problem or use case? Please describe.**
We need to add a few improvements to the datepicker so that it can be used standalone and in a smaller version.

**Describe the solution you'd like**
- card popover background is removed to do this https://master-enterprise.demo.design.infor.com/components/monthview can be used already but we need to make the monthview that is in https://master-enterprise.demo.design.infor.com/components/datepicker/example-index avaialble "in page". There is an option now for it being in a datepicker but need one so the smaller one there is shown in page. Would add an example to components/monthview/example-inpage . And the example should look like the design
- Clear and Apply actions are removed to do this just dont include buttons in the example-inpage version
- Tapping a date can trigger and event that changes the list below  (application team’s responsibility but we add the event)
~- Add swiping in our Angular web component (and HTML??)~ this will be done on https://github.com/infor-design/enterprise/issues/5002
- Change “Today” action to 16px bold
- Spacing between dates and chevrons has been improved to have enough space for a 40 to 44px hitbox (for in page version only if this is easier / breaking too much / this is optional)
- Change date hitbox to 40 or 44px for better mobile touch points (if space allows) (for in page version only if this is easier / breaking too much / this is optional)
- Included a legend view from the datepicker component, can we align it to the left by default? Centered is ok for the popover examples. Suggest we do this based on number of items. If they can fit in one row do it left aligned otherwise centered
- If clicked/tapped, the toolbar title & calendar icon will pulse & expand the month view downwards... pushing other content down the page. To do this we do it in the in page example
- Change text color for current month’s dates to slate 8 (can change all)
- azure and slate circles are 2 instead of 1. To do this - its just part of the example in the legend colors
- Add ‘pulse’ animation back in for buttons/dates on mobile at least?
- Noticed that on https://main-enterprise.demo.design.infor.com/components/monthview/example-index when you click the calendar it changes the month this may be related to https://github.com/infor-design/enterprise/issues/5064
- Adjust the month and year picker

**Describe alternatives you've considered**
Leaving this go for now

**Additional context**
See designs - we are just implementing page one (High Priority) https://www.figma.com/file/ri2Knf3KchdfdzRAeds0Ab/IDS-Mobility-v4.6?node-id=1%3A5739

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
- Added support for mobile view with Monthview.
- Fixed selected date was not stay on provided day/month/year.

**Related github/jira issue (required)**:
Closes #5075
Closes #5064

**Steps necessary to review your pull request (required)**:
Pull this branch and build/run the demo app

Test for 5075
- Navigate to:
- http://localhost:4000/components/monthview/example-inpage.html
- http://localhost:4000/components/monthview/test-inpage-variations.html
- Open developer tools
- Check the in-page styles [here](https://www.figma.com/file/ri2Knf3KchdfdzRAeds0Ab/IDS-Mobility-v4.6?node-id=1%3A5739)
- Check for Collapse, Hide Today, Hide Month Year Picker, Legend and No Toggleable
- If clicked the toolbar title or calendar icon will pulse and expand and pushing other content down the page.
- Text color for current month’s dates should be slate-8
- See each date hitbox minimum 40px
- Should be `pulse` animation back in for buttons and dates
- Legends, if single line should be left align and wrapped should center alignment
- Check the style for month and year picker

Test for 5064
- Navigate to: http://localhost:4000/components/calendar/test-provided-date.html
- See the selected date should be 4/21/2021

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

File: src/components/calendar-toolbar/calendar-toolbar.js [hunk 1]
```diff
@@ -41,7 +44,9 @@ const COMPONENT_DEFAULTS = {
   isMenuButton: true,
   showViewChanger: false,
   viewChangerValue: 'month',
-  isMonthPicker: false
+  isMonthPicker: false,
+  inpage: false,
```
Comment (tmcconechy) [inline src/components/calendar-toolbar/calendar-toolbar.js:48]: Suggest we call it inPage but thats very minor

**Review by tmcconechy — CHANGES_REQUESTED** [review]
This is looking really nice just some small suggestions

- instead of inpage call it inPage in the settings
- "No Toggleable" -> "Not Toggleable"
- the toolbar looks a little out of alignment
Screen Shot 2021-05-26 at 9 41 51 AM
- seeing an extra scrollbar
Screen Shot 2021-05-26 at 9 42 38 AM
- the colors we use in the legend are quite hard to see can we bring them each up one in the color scale?
Screen Shot 2021-05-26 at 9 46 29 AM

**Comment by deep7102** [comment]
@tmcconechy Updated now, for all the points you see and there is no markup change only the api settings.

**Comment by deep7102** [comment]
It's ready to review now

**Comment by tmcconechy** [comment]
Just one thing.... It looks like now the icon is quite far away?

Screen Shot 2021-05-27 at 10 18 26 AM

**Comment by tmcconechy** [comment]
@kevinwhitedesign and had some comments... 

- [x] Remove the tags
- [ ] Make header sticky and scrollable

I fixed the first one but the second one i will do when we add swipe and this is a seperate ticket https://github.com/infor-design/enterprise-wc/issues/133

Please review this live links and if any comments on this please do let us know here:

https://5075-datepicker-mobility-enhancements-enterprise.demo.design.infor.com/components/monthview/test-inpage-variations.html
https://5075-datepicker-mobility-enhancements-enterprise.demo.design.infor.com/components/monthview/example-inpage.html

cc @benfenton

**Comment by deep7102** [comment]
@tmcconechy Updated now, spacing icon vs title.

**Review by tmcconechy — APPROVED** [review]

**Comment by ericangeles** [comment]
Very nice work, @deep7102! Just noticed a small issue on mobile viewport `320px`. If the month text is too long, some of the components gets cut off. Both classic and new themes.


Screen Shot 2021-05-31 at 11 44 13 PM

The right carat is missing, and the month text is cut off.

Am okay to pass this and fix it some other time.

**Review by EdwardCoyle — COMMENTED** [review]
@deep7102 Everything looks good here!  I saw what @ericangeles mentioned as well but seems to be OK on most iOS simulators (see below).  I'd be OK punting on it for now

Screen Shot 2021-06-01 at 9 46 27 AM

**Review by EdwardCoyle — APPROVED** [review]

**Review by ericangeles — APPROVED** [review]

**Pull request merged**